src/main/jit/Optimization.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ *  R : A Computer Language for Statistical Data Analysis
+ *  Copyright (C) 2014 and onwards the Rho Project Authors.
+ *
+ *  Rho is not part of the R project, and bugs and other issues should
+ *  not be reported via r-bugs or other R project channels; instead refer
+ *  to the Rho website.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2.1 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, a copy is available at
+ *  http://www.r-project.org/Licenses/
+ */
+
+#include "rho/jit/Optimization.hpp"
+
+#include <string>
+#include <vector>
+
+#include "rho/jit/llvm.hpp"
+
+namespace rho {
+namespace JIT {
+namespace {
+const std::string kSetVisibilityFuncName("rho_runtime_setVisibility");
+}  // namespace
+
+bool BasicFunctionPass::runOnFunction(llvm::Function& function) {
```

Comment:
That's a good idea. I will make it into a separate basic block pass.  I was thinking of expanding this into a function-level redundancy elimination pass later, but at that point I can always convert it to a function-level pass.